From d73ad8cbc96d815b0eca244ae363957b0c024acb Mon Sep 17 00:00:00 2001
From: Jurijs Soloveckis <jsoloveckis@maxlinear.com>
Date: Mon, 25 Mar 2024 13:23:12 +0000
Subject: [PATCH] Enable beacon protection if IEEE 802.11be/EHT is enabled for
 BSS

IEEE P802.11be/D7.0, 12.12.9 (Security constraints for EHT) mandates
beacon protection to be enabled in EHT AP when using RSN. Do this
automatically even if beacon_prot=1 is not included in the
configuration.

Signed-off-by: Jurijs Soloveckis <jsoloveckis@maxlinear.com>
---
 src/ap/ap_config.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/ap/ap_config.c b/src/ap/ap_config.c
index 160809a27..a68454c01 100644
--- a/src/ap/ap_config.c
+++ b/src/ap/ap_config.c
@@ -1511,6 +1511,13 @@ static int hostapd_config_check_bss(struct hostapd_bss_config *bss,
 		wpa_printf(MSG_INFO,
 			   "Disabling IEEE 802.11be as IEEE 802.11ax is disabled for this BSS");
 	}
+
+	if (full_config && conf->ieee80211be && !bss->disable_11be &&
+	    !bss->beacon_prot && ap_pmf_enabled(bss)) {
+		bss->beacon_prot = 1;
+		wpa_printf(MSG_INFO,
+			   "Enabling beacon protection as IEEE 802.11be is enabled for this BSS");
+	}
 #endif /* CONFIG_IEEE80211BE */
 
 	if (full_config && bss->ignore_broadcast_ssid && conf->mbssid) {
-- 
2.43.0

